src/LanguageServer/Handler/Commands.hs

```diff
@@ -0,0 +1,63 @@
+-----------------------------------------------------------------------------
+-- The LSP handlers that handles initialization
+-----------------------------------------------------------------------------
+{-# LANGUAGE OverloadedStrings #-}
+
+module LanguageServer.Handler.Commands (initializedHandler, commandHandler) where
+
+import Compiler.Options (Flags (outFinalPath), targets, commandLineHelp, updateFlagsFromArgs)
+import Language.LSP.Server (Handlers, LspM, notificationHandler, sendNotification, MonadLsp, getVirtualFiles, withIndefiniteProgress, requestHandler)
+import qualified Language.LSP.Protocol.Types as J
+import qualified Data.Text as T
+import LanguageServer.Monad (LSM, getFlags, getTerminal)
+import qualified Language.LSP.Protocol.Message as J
+import Data.Aeson as Json
+import qualified Language.LSP.Protocol.Lens as J
+import Control.Lens ((^.))
+import Data.Maybe (mapMaybe, fromJust, fromMaybe)
+import GHC.Base (Type)
+import LanguageServer.Handler.TextDocument (recompileFile)
+import Compiler.Compile (CompileTarget(..), Terminal (termError, termPhaseDoc))
+import Common.Name (newName)
+import qualified Language.LSP.Server as J
+import Control.Monad.Trans (liftIO)
+
+initializedHandler :: Handlers LSM
+initializedHandler = notificationHandler J.SMethod_Initialized $ \_not -> sendNotification J.SMethod_WindowLogMessage $ J.LogMessageParams J.MessageType_Info "Initialized language server."
+
+targetFlag :: String -> Flags -> Flags
+targetFlag t f
+    = case lookup t targets of
+        Just update -> update f
+        Nothing     -> f
+
+commandHandler :: Handlers LSM
+commandHandler = requestHandler J.SMethod_WorkspaceExecuteCommand $ \req resp -> do
+  flags <- getFlags
+  let J.ExecuteCommandParams _ command commandParams = req ^. J.params
+  if command == "koka/genCode" then
```

Comment:
This is the genCode command, which instructs the language server to actually output an executable and report the executable's location